### DISCUSSION TITLE
Is it possible to swap out an Immich server but keep the old image file system/structure?

### DISCUSSION THREAD

**Discussion by GuoxiongXie** [immich-app/immich#3952] [Q&A] [answered]
Hi Immich family, hope all is well. I'm new to this amazing Immich, so please bear with me if my question doesn't quite make sense.
I installed Immich by following this recommended way: [Docker Compose](https://immich.app/docs/install/docker-compose).
My UPLOAD_LOCATION is pointing to an external hard drive. After the installation, I do see Immich correctly stores photos in file structure in my external hard drive like `UPLOAD_LOCATION/library/admin/2023/2023-08-30`.

After some use, I feel like that I need a more powerful machine to run Immich. So, I'm wondering if I can follow [Docker Compose](https://immich.app/docs/install/docker-compose) again to install Immich app on my new machine and then point UPLOAD_LOCATION to the same external hard drive where photos are uploaded by the Immich app on my old machine. Would the Immich app on my new machine be able to read photos from the external hard drive automagically without me uploading those files again?
I suspect that I might need to [restore Immich database from a database dump](https://immich.app/docs/administration/backup-and-restore) if I don't want to upload the photos again. Is this true? 

Thank you very much for your help!

**Comment by alextran1502** [comment, marked as answer]
Hello, your thoughts are correct, you will need to move the database to the new machine so that it knows where to get the files from the UPLOAD_LOCATION

**Reply by jrasm91** [reply]
You can also reference this:

https://immich.app/docs/administration/backup-and-restore
